reactor-netty-core/src/main/java/reactor/netty/tcp/SslProvider.java

```diff
@@ -719,6 +751,14 @@ public Builder setSniMappings(Map<String, Consumer<? super SslContextSpec>> conf
 			Objects.requireNonNull(confPerDomainName);
 			this.confPerDomainName.clear();
 			confPerDomainName.forEach(this::addInternal);
+			this.sniMappings = null;
+			return this;
+		}
+
+		@Override
+		public Builder setSniMappings(AsyncMapping<String, SslProvider> mappings) {
```

Comment:
consider maybe renaming it slightly to indicate this is not entirely compatible with the legacy methods? something like `setSniAsyncMappings`?

Reply:
fixed